src/mono/mono/mini/interp/interp-simd.c

```diff
@@ -0,0 +1,605 @@
+
+#include "interp-internals.h"
+#include "interp-simd.h"
+
+#ifdef INTERP_ENABLE_SIMD
+
+#define SIZEOF_V128	16
+
+typedef gint64 v128_i8 __attribute__ ((vector_size (16)));
+typedef guint64 v128_u8 __attribute__ ((vector_size (16)));
+typedef gint32 v128_i4 __attribute__ ((vector_size (16)));
+typedef guint32 v128_u4 __attribute__ ((vector_size (16)));
+typedef gint16 v128_i2 __attribute__ ((vector_size (16)));
+typedef guint16 v128_u2 __attribute__ ((vector_size (16)));
+typedef gint8 v128_i1 __attribute__ ((vector_size (16)));
+typedef guint8 v128_u1 __attribute__ ((vector_size (16)));
+
+// get_AllBitsSet
+static void
+interp_v128_i4_all_bits_set (gpointer res)
+{
+	memset (res, 0xff, SIZEOF_V128);
+}
+
+// op_Addition
+static void
+interp_v128_i1_op_addition (gpointer res, gpointer v1, gpointer v2)
+{
+	*(v128_i1*)res = *(v128_i1*)v1 + *(v128_i1*)v2;
+}
+
+static void
+interp_v128_i2_op_addition (gpointer res, gpointer v1, gpointer v2)
+{
+	*(v128_i2*)res = *(v128_i2*)v1 + *(v128_i2*)v2;
+}
+
+static void
+interp_v128_i4_op_addition (gpointer res, gpointer v1, gpointer v2)
+{
+	*(v128_i4*)res = *(v128_i4*)v1 + *(v128_i4*)v2;
+}
+
+// op_Subtraction
+static void
+interp_v128_i1_op_subtraction (gpointer res, gpointer v1, gpointer v2)
+{
+	*(v128_i1*)res = *(v128_i1*)v1 - *(v128_i1*)v2;
+}
+
+static void
+interp_v128_i2_op_subtraction (gpointer res, gpointer v1, gpointer v2)
+{
+	*(v128_i2*)res = *(v128_i2*)v1 - *(v128_i2*)v2;
+}
+
+static void
+interp_v128_i4_op_subtraction (gpointer res, gpointer v1, gpointer v2)
+{
+	*(v128_i4*)res = *(v128_i4*)v1 - *(v128_i4*)v2;
+}
+
+// op_BitwiseAnd
+static void
+interp_v128_op_bitwise_and (gpointer res, gpointer v1, gpointer v2)
+{
+	*(v128_i4*)res = *(v128_i4*)v1 & *(v128_i4*)v2;
+}
+
+// op_BitwiseOr
+static void
+interp_v128_op_bitwise_or (gpointer res, gpointer v1, gpointer v2)
+{
+	*(v128_i4*)res = *(v128_i4*)v1 | *(v128_i4*)v2;
+}
+
+// op_Division
+static void
+interp_v128_i1_op_division (gpointer res, gpointer v1, gpointer v2)
+{
+	*(v128_i1*)res = *(v128_i1*)v1 / *(v128_i1*)v2;
+}
+
+static void
+interp_v128_u1_op_division (gpointer res, gpointer v1, gpointer v2)
+{
+	*(v128_u1*)res = *(v128_u1*)v1 / *(v128_u1*)v2;
+}
+
+static void
+interp_v128_i2_op_division (gpointer res, gpointer v1, gpointer v2)
+{
+	*(v128_i2*)res = *(v128_i2*)v1 / *(v128_i2*)v2;
+}
+
+static void
+interp_v128_u2_op_division (gpointer res, gpointer v1, gpointer v2)
+{
+	*(v128_u2*)res = *(v128_u2*)v1 / *(v128_u2*)v2;
+}
+
+static void
+interp_v128_i4_op_division (gpointer res, gpointer v1, gpointer v2)
+{
+	*(v128_i4*)res = *(v128_i4*)v1 / *(v128_i4*)v2;
+}
+
+static void
+interp_v128_u4_op_division (gpointer res, gpointer v1, gpointer v2)
+{
+	*(v128_u4*)res = *(v128_u4*)v1 / *(v128_u4*)v2;
+}
+
+// op_Equality
+static void
+interp_v128_op_equality (gpointer res, gpointer v1, gpointer v2)
```

Comment:
Should this qualify that it's bitwise equality?

It will be incorrect for floating-point if that's added in the future, for example